src/compiler/parser.ts

```diff
@@ -542,6 +524,16 @@ namespace ts {
         let TokenConstructor: new (kind: SyntaxKind, pos: number, end: number) => Node;
         let IdentifierConstructor: new (kind: SyntaxKind, pos: number, end: number) => Node;
         let SourceFileConstructor: new (kind: SyntaxKind, pos: number, end: number) => Node;
+        interface Fail extends Node { kind: SyntaxKind.Unknown; }
+        interface FailList extends NodeArray<Node> { pos: -1; }
+        let Fail: Fail;
+        let FailList: FailList;
+        function isFail(x: Node | undefined): x is Fail {
+            return !!x && x.kind === SyntaxKind.Unknown;

```

Comment:
`return Fail !== undefined && x === Fail;` doesn't require the property lookup and verifies that `Fail` has a value.